components/public-api-server/pkg/oidc/state_jwt.go

```diff
@@ -11,17 +11,13 @@ import (
 )
 
 type StateClaims struct {
-	// Internal client ID
-	ClientConfigID string `json:"clientId"`
-	ReturnToURL    string `json:"returnTo"`
-
+	StateParams StateParams `json:"stateParams"`
```

Comment:
What's the reason for the deeper nesting? We're unlikely to have more properties in this StateClaims object given it already encodes the state, so the extra StateParams feels redundant. What if we removes StateParams and used StateClaims in other places instead?

Reply:
Good question. The problem is, I wanted to have a superset of `jwt.RegisteredClaims` with the additional properties from `StateParams`, while being able to use `StateParams` as a type in other places. The motivation to have a coherent object popped up when I found the bug that the `Activated` property added here was not propagated, a typical mapping bug between `StateParams` and `StateClaims` basically.

> We're unlikely to have more properties in this StateClaims object given it already encodes the state, so the extra StateParams feels redundant.

Don't think so. `StateParams` are describing options requested by an anonymous user, which would be present once the identity of the user is received. That would give us a chance to pass more properties on this journey. 

Just to be sure, the nesting isn't ideal from my pov. Ideally would have use struct embedding for both:

```
type StateClaims struct {
	StateParams
	jwt.RegisteredClaims
}
```

If you have an idea, how to fix this 🙏🏻

Reply:
Sure, let's do the embedding- that should make the claims flat, and ensures all the fields are present, no?

Reply:
That's what I'm saying, @easyCZ, embedding of the two of them doesn't work as expected, it still produces the same structure. I couldn't find a reason. If you embed a single one with additional props, it's flat, but adding a second produces this kind of structure anyways. 

I'm happy to change to a better version. Error prone prop copying is what I meant to avoid here, as it's already caused a bug. If it's ok, Iet's do a follow-up on this.

Reply:
Looking at this, I still feel we're better off flattening it. Yes, the conversion can be error prone but to aid with that you can go through a constructor function which has fixed set of arguments (and here, we don't have that many of them). But it's not worth blocking on this, and given the tokens are effectively use-once, changing this doesn't even need to be backwards compatible should we want to do it.